tests/foreman/ui/test_adusergroups.py

```diff
@@ -0,0 +1,134 @@
+"""Test class for Active Directory Feature"""
+from robottelo.test import UITestCase
+
+
+class ADUserGroups(UITestCase):
+    """Implements Active Directory feature tests in UI."""
+
+    def test_create_ldap_auth_withAD(self):
+        """@Test: Create LDAP authentication with AD
+
+        @Feature: Active Directory - create LDAP AD
+
+        @steps:
+
+        1. Create a new LDAP Auth source with AD
+
+        2. Fill in all the fields appropriately for AD
+
+        @Assert: Whether creating LDAP Auth with AD is successful
+
+        """
+
+    def test_adusergroup_admin_role(self):
+        """@Test: Associate Admin role to User Group.
+        [belonging to external AD User Group.]
+
+        @Feature: Active Directory - associate Admin role
+
+        @Steps:
+
+        1. Create an UserGroup.
+
+        2. Assign admin role to UserGroup.
+
+        3. create and associate an External AD UserGroup.
+
+        @Assert: Whether a User belonging to User Group is able to
+        access all the pages.
+
+        """
+
+    def test_adusergroup_foreman_role(self):
+        """@Test: Associate foreman roles to User Group.
+        [belonging to external AD User Group.]
+
+        @Feature: Active Directory - associate foreman roles.
+
+        @Steps:
+
+        1. Create an UserGroup.
+
+        2. Assign some foreman roles to UserGroup.
+
+        3. create and associate an External AD UserGroup.
+
+        @Assert: Whether a User belonging to User Group is able to
+        access foreman entities as per roles.
+
+        """
+
+    def test_adusergroup_katello_role(self):
+        """@Test: Associate katello roles to User Group.
+        [belonging to external AD User Group.]
+
+        @Feature: Active Directory - associate katello roles
+
+        @Steps:
+
+        1. Create an UserGroup.
+
+        2. Assign some foreman roles to UserGroup.
+
+        3. create and associate an External AD UserGroup.
+
+        @Assert: Whether a User belonging to User Group is able to access
+        katello entities as per roles.
+
+        """
+
+    def test_create_external_adusergroup_1(self):
+        """@Test: Create External AD User Group as per AD group
+
+        @Feature: Active Directory - create
+
+        @Steps:
+
+        1. Create an UserGroup.
+
+        2. Assign some foreman roles to UserGroup.
+
+        3. create an External AD UserGroup as per the usergroup name in AD
+
+        @Assert: Whether creation of External AD User Group is possible.
+
+        """
+
+    def test_create_external_adusergroup_2(self):
+        """@Test: Create another External AD User Group with same name
+
+        @Feature: Active Directory - create
+
+        @Steps:
+
+        1. Create an UserGroup.
+
+        2. Assign some foreman roles to UserGroup.
+
+        3. create an External AD UserGroup as per the usergroup name in AD
+
+        4. Repeat steps 1) and 2) and provide the same usergroup name for
+        step 3)
```

Comment:
Make sure to evenly indent all text that follows a list marker.

    4. Repeat steps 1) and 2) and provide the same usergroup name for
       step 3)